### PR TITLE
Reviewed exceptions in graphql-java

### PR DESCRIPTION
**Pull request by bbakerman**
See #751 

I went through a reviewed the exceptions that bubble all the way up to the top of graphql.

I documented them in the JavaDoc and I think I will continue this and document them in .rts as well.

With graphql-java (and graphql in general) is not clear what is an error and whats an exception.

There seems like a lot of changes here but there really isnt.

Also I worked out how to build and run the local docs html

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (andimarek) [inline docs/Makefile]: just as a note for future improvements: we should also really document how to build it via Docker.

File: docs/exceptions.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
+Runtime Exceptions
+==================
+
```
Comment (andimarek) [inline docs/exceptions.rst]: We should document what it means 'can be thrown by the graphql engine'. Does it mean I get an exception in `GraphQL.execute' or a special error etc ...

Comment (bbakerman) [reply docs/exceptions.rst]: updated

File: docs/exceptions.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Runtime exceptions can be thrown by the graphql engine if certain exceptional situations are encountered.  The following
+are a list of the exceptions that can be thrown all the way out of a ``graphql.execute(...)`` call.
+
+These are not graphql errors in execution but rather totally unacceptable conditions in which to execute a graphql query.
+ 
+ -  `graphql.schema.CoercingSerializeException`
```
Comment (andimarek) [inline docs/exceptions.rst]: I don't think this exception can escalate to the end user a exception thrown from `.execute` ... we handle it in `handleCoercionProblem`

**Pull request merged**